_pages/doc/contributing-to-llvm.md

```diff
@@ -0,0 +1,168 @@
+
+```sh
+cmake \
+    -B build \
+    -D BUILD_SHARED_LIBS="On" \
+    -D CMAKE_BUILD_TYPE="Release" \
+    -D LLVM_ENABLE_ASSERTIONS="On" \
+    -D LLVM_ENABLE_PROJECTS="" \
+    -D LLVM_EXPERIMENTAL_TARGETS_TO_BUILD="M68k" \
+    -D LLVM_TARGETS_TO_BUILD="" \
+    -D LLVM_USE_SPLIT_DWARF="On" \
+    -G Ninja \
+    -S llvm
+```
+
+#### Rationale for the CMake options
+
+CMake usage can be displayed with `cmake --help`.
+
+The LLVM CMake options which affect the build are [documented here](https://llvm.org/docs/CMake.html).
+
+`-B build`
+
+We'll use a directory named `build` to store build files. This directory is already ignored by the `.gitignore` file, so it's assumed to be the standard build directory.
+
+`-D BUILD_SHARED_LIBS="On"`
+
+We'll build shared libraries rather than static libraries, to save some disk space.
+
+`-D CMAKE_BUILD_TYPE="Release"`
+
+We'll build the `Release` configuration, as otherwise running tests can be quite slow.
+
+**TODO:** Should we build the `RelWithDebInfo` configuration instead?
+
+`-D LLVM_ENABLE_ASSERTIONS="On"`
+
+We'll enable assertions, in order to catch more errors during runtime.
+
+`-D LLVM_ENABLE_PROJECTS=""`
+
+We'll remove unnecessary projects from the build.
+
+`-D LLVM_EXPERIMENTAL_TARGETS_TO_BUILD="M68k"`
+
+The M68k target is "experimental", meaning it's not built by default. We need to explicitly enable it.
+
+`-D LLVM_TARGETS_TO_BUILD=""`
+
+We'll remove all other targets from the build.
+
+`-D LLVM_USE_SPLIT_DWARF="On"`
+
+Remove memory pressure during link-time.
+
+`-G Ninja`
+
+Use the Ninja build system, rather than the default (Make.) Ninja is faster than Make.
+
+`-S llvm`
+
+Specify `llvm` as the source directory.
+
+## Building LLVM for testing
+
+Invoke CMake to build:
+
+```sh
+cmake --build build --target test-depends
+```
+
+The `test-depends` target contains all the dependencies required to run tests.
+
+## Running the M68k tests
+
+It takes significantly longer to run all the tests than it does to just run the M68k-related tests. In order to iterate more quickly, you can run only the M68k tests, rather than all the tests.
+
+Invoke `llvm-lit` to run the M68k tests:
+
+```sh
+./build/bin/llvm-lit \
+    llvm/test/CodeGen/M68k \
+    llvm/test/DebugInfo/M68k \
+    llvm/test/MC/Disassembler/M68k \
+    llvm/test/MC/M68k
+```
+
+`llvm-lit` is the [LLVM Integrated Tester](https://llvm.org/docs/CommandGuide/lit.html) utility.
+
+We specify the directories containing M68k-related tests.
+
+You should see output something like:
+
+```
+-- Testing: 168 tests, 8 workers --
+PASS: LLVM :: CodeGen/M68k/Arith/lshr.ll (1 of 168)
+PASS: LLVM :: CodeGen/M68k/Alloc/dyn_alloca_aligned.ll (2 of 168)
+(more tests)
+PASS: LLVM :: MC/M68k/pc-rel.s (168 of 168)
+
+Testing Time: 7.61s
+
+Total Discovered Tests: 168
+  Passed: 168 (100.00%)
+```
+
+## Running all the tests
+
+Invoke CMake to run all the tests:
+
+```sh
+cmake --build build --target check-all
```

Comment:
we can also use `check-llvm-codegen-m68k` to run all tests under `test/CodeGen/M68k` -- LLVM's cmake system actually generates a bunch of build targets according to the test folder hierarchy. So similarly, `check-llvm-mc-m68k` will run every tests under `test/MC/M68k` and `check-llvm-mc-disassembler-m68k` for tests under `test/MC/Disassembler/M68k`